scripts/task_completion_tracker.py

```
@@ -0,0 +1,407 @@
+                record for record in self.completion_records
+                if record.end_time >= cutoff_time and record.success and
+                (agent_id is None or record.agent_id == agent_id) and
+                (task_type is None or record.task_type == task_type)
+            ]
+            
+            if not relevant_records:
+                return 0.0
+                
+            return statistics.mean(record.duration for record in relevant_records)
+            
+    def get_completion_trends(self, agent_id: str = None, task_type: str = None,
```

Ruff (0.14.2)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)